django-stubs/core/files/base.pyi

```diff
@@ -26,7 +26,7 @@ class File(FileProxyMixin[AnyStr], IO[AnyStr]):
         exc_value: BaseException | None,
         exc_tb: TracebackType | None,
     ) -> None: ...
-    def open(self, mode: str | None = ...) -> Self: ...
+    def open(self, mode: str | None = ..., *args: Any, **kwargs: Any) -> Self: ...
```

Comment:
Since args+kwargs are passed to Python `builtins.open()`, we can copy more precise types from there.

https://github.com/python/typeshed/blob/7acaa912b042e5aa28f679748c94f587da99f846/stdlib/builtins.pyi#L1553-L1564

```suggestion
    def open(
        self,
        mode: str | None = ...,
        buffering: int = -1,
        encoding: str | None = None,
        errors: str | None = None,
        newline: str | None = None,
        closefd: bool = True,
        opener: Callable[[str, int], int] | None = None,
    ) -> Self: ...
```